dan_layer/core/src/services/asset_processor.rs

```diff
@@ -89,12 +89,12 @@ impl AssetProcessor for ConcreteAssetProcessor {
         )
     }
 
-    fn invoke_read_method<TUnifOfWork: StateDbUnitOfWork>(
+    fn invoke_read_method<TUnitOfWork: StateDbUnitOfWorkReader>(
```

Comment:
Nice. Perhaps we need a better name, but I like it

Reply:
Yeah it's a bit long, open to suggestions/change later